app/controllers/application_controller.rb

```diff
@@ -68,7 +68,22 @@ def set_locale
     m_community = Maybe(@current_community)
     community_locales = m_community.locales.or_else([])
     community_default_locale = m_community.default_locale.or_else("en")
-    # We should remove this -- END
+
+    # Load translations from TranslationService
+    community_backend = I18n::Backend::CommunityBackend.instance
+    m_community.map(&:id).each { |community_id|
+      community_backend.community_id = community_id
+      community_translations = TranslationService::API::Api.translations.get(community_id)[:data]
+      TranslationServiceHelper.community_translations_for_i18n_backend(community_translations).each { |locale, data|
+        # Store community translations to I18n backend.
+        #
+        # Since the data in data hash is already flatten, we don't want to
+        # escape the separators (. dots) in the key
+        community_backend.store_translations(locale, data, escape: false)
```

Comment:
Storing new translations to the backend per each request will grow the translation hash without ever being cleaned up. Potentially, this may be an issue.